src/syft/core/node/common/action/function_or_constructor_action.py

```diff
@@ -104,7 +104,24 @@ def execute_action(self, node: AbstractNode, verify_key: VerifyKey) -> None:
             )
             resolved_kwargs[arg_name] = r_arg.data
 
-        result = method(*resolved_args, **resolved_kwargs)
+        # when we call original constructors like torch.device we will get errors
```

Comment:
looks great!

also - didn't know this was called "upcast"ing - im a fan

Reply:
Yeah I had to google it to double check... I was like... its not down casting... so whats the opposite?
There is probably a better mathy name for it somewhere like something to do with "contravariance" or something.